sdk/src/Core/Amazon.Runtime/BearerTokenSigner.cs

```diff
@@ -43,16 +48,12 @@ public override void Sign(
                     "Endpoint must not use 'http'.");
             }
 
-            if (null == clientConfig.AWSTokenProvider)
-                return;
-
-            if (!clientConfig.AWSTokenProvider.TryResolveToken(out var token) ||
-                string.IsNullOrEmpty(token.Token))
+            if (string.IsNullOrEmpty(ResolvedToken))
             {
                 throw new AmazonClientException("No Token found.  Operation requires a Bearer token.");
             }
 
-            request.Headers["Authorization"] = $"Bearer {token}";
+            request.Headers["Authorization"] = $"Bearer {ResolvedToken}";
         }
 #else
         public override void Sign(
```

Comment:
Yes, this was a miss on my part. The conditional compilations were required before because this class was using the token provider implementation directly. It's not needed anymore now that the token is provided differently.